File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/c/non-vul/code_before33.c
// From cwe-snippets, snippets_90/non-compliant/C/cwe-0190/char_fscanf_multiply_01.c

static void func()
{
    char data;
    data = ' ';
    data = 2;
    if(data > 0) /* ensure we won't have an underflow */
    {
        char result = data * 2;
        printHexCharLine(result);
    }
}

File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/java/non-vul/code_before33.java
// From cwe-snippets, snippets_90/non-compliant/Java/cwe-0015/connect_tcp_03.java

private void func() throws Throwable
    {
        String data;
        if (5 == 5)
        {
            data = "foo";
        }
        else
        {
            data = null;
        }

        Connection dbConnection = null;

        try
        {
            dbConnection = IO.getDBConnection();
            dbConnection.setCatalog(data);
        }
        catch (SQLException exceptSql)
        {
            IO.logger.log(Level.WARNING, "Error getting database connection", exceptSql);
        }
        finally
        {
            try
            {
                if (dbConnection != null)
                {
                    dbConnection.close();
                }
            }
            catch (SQLException exceptSql)
            {
                IO.logger.log(Level.WARNING, "Error closing Connection", exceptSql);
            }
        }

    }